devito/passes/iet/misc.py

```diff
@@ -204,3 +207,42 @@ def remove_redundant_moddims(iet):
     iet = Transformer(subs, nested=True).visit(iet)
 
     return iet
+
+
+def abridge_dim_names(iet):
+    def replace_parent(dim, mapper):
+        return dim._rebuild(parent=mapper[dim.parent])
+
+    subs = {}
+    mapper = {}
+    umapper = {}
+    tree = retrieve_iteration_tree(iet)
+    for it in tree:
+        indexeds = FindSymbols('indexeds').visit(it.inner)
+        inds = set().union(*[pull_dims(i, flag=False) for i in indexeds])
+        inds = [i for i in inds if any([d.is_Sub for d in i._defines])]
+        inds = [i for i in inds if not i.is_SubIterator]
+        names = [i.root.name for i in inds]
+
+        mapper.update({i: i._rebuild(i.root.name) for i in inds
+                       if i.root not in it.dimensions and names.count(i.root.name) < 2})
+
+    if mapper:
+        for it in tree:
+            uinds = set().union(*[i.uindices for i in it])
+            uinds = [i for i in uinds if i.is_Incr]
+            uinds = [i for i in uinds if i.parent in mapper]
+            umapper.update({i: replace_parent(i, mapper) for i in uinds})
+
+        for it in tree:
+            mdims = FindSymbols('dimensions').visit(it)
+            mdims = [d for d in mdims if d.is_Modulo]
+            mdims = [d for d in mdims if d.parent in {**mapper, **umapper}]
+            mmapper = {d: replace_parent(d, {**mapper, **umapper}) for d in mdims}
```

Comment:
either create that combined dictioanry once or call `replace_parent` separately for each one